### DISCUSSION TITLE
Feature Request: Document-Level Permissions

### DISCUSSION THREAD

**Discussion by mulder999** [outline/outline#7136] [Ideas]
## Current Functionality

As-is, the system allows setting permissions at the collection level. This includes the ability to create documents and nested documents  within these collections.

## Requested improvement

The system should also support adding more restrictive permissions at the individual document and nested document levels.

## Benefits

Implementing document-level permissions would enable more granular control over access rights, improving security and collaboration capabilities.

## Use Case Example

- Collection `Projects` (granted to team `T1`)
  - Document `Project A` (permissions inherited from `Projects`)
  - Document `Project Super Secret` (granted to team `T2`, limiting access to `T1 and T2`)
    - Nested document `Some info for project super secret` (permissions inherited from `Project Super Secret`)

Only members of `T2` who are also part of `T1` could access `Project Super Secret` and below.

**Comment by bladerunner2020** [comment]
We also badly miss this feature. We migrated to Outline from Notion, where we could set up group-based permissions. Setting permissions individually is far from convenient when managing more than a few people.

**Reply by tommoor** [reply]
Outline includes permissions for both users and groups – this was added in September of '24:

https://www.getoutline.com/changelog/group-document-membership

**Reply by thomas-zic** [reply]
> Only members of `T2` who are also part of `T1` could access `Project Super Secret` and below.

In my opinion, this is an essential feature!

Perhaps it's already possible, without removing everyone's access to the collection of course. The idea is to keep the collection tree structure but hide certain documents from certain groups/users. 

How can this be achieved with the current version of Outline (0.81.1)?

**Reply by tommoor** [reply]
It is not possible, and it's unlikely to be so in the near future as it would require a rethinking of the permissions system which is always additive for performance and usability reasons. If this is important for your team I recommend considering other software

**Reply by thomas-zic** [reply]
I understand that managing permissions by document is a new thing in Outline and we're keen to get involved in your project by making full use of it, even if it means paying for a licence.

You must understand that in the interests of confidentiality within the same team, not all members should be able to access the same resources through a project. And it would be overkill to have to create a new collection just to share a document that must not be revealed to members who must not know about it. Even more so when there is a logic with the nesting of documents.

We really like the tree system, but when you share documents with a member who doesn't have access to the collection, they end up in a ‘Shared with me’ section and have no way of knowing the context without opening it (unless you include the name of the collection in the name of the document, but that would also be overkill).

At least there should be an option in a document's permissions to restrict access to certain groups/users, otherwise the permissions inherit those of the parent document. Then there would be no need to restrict access to the entire collection.

Our teams are far from being the only ones interested in such a feature, even if they have to pay for it, because data confidentiality is at stake. Don't you agree with this idea?

**Reply by bladerunner2020** [reply]
> Outline includes permissions for both users and groups – this was added in September of '24:
> 
> 
> 
> https://www.getoutline.com/changelog/group-document-membership

Thank you very much! I’m not sure how I overlooked it—I thought I had checked all the settings (though clearly not all) before making my comment.

We have an on-premises installation, but the groups are there as well. I’ll have our DevOps team integrate them with Keycloak. 👍🏻

**Reply by aaran-stent** [reply]
> It is not possible, and it's unlikely to be so in the near future as it would require a rethinking of the permissions system which is always additive for performance and usability reasons. If this is important for your team I recommend considering other software

Hello, we have deployed Outline Enterprise to manage our IT documentation including our Vendors and their contracts.  I would like to be able to remove permission inheritance so that some pages with pricing can be blocked from being visible to all 70 IT staff and limited to staff that need to know the licensing model and its pricing.  How can I upvote and/or sponsor this feature?

The only workaround I have for now is to leave the pricing data off Outline and link it (we use SharePoint) to achieve this.  Or create a new collection and move pricing data to that collection with limited access to NDA signatories.
